File: updated_code/code/java/1446-consecutive-characters.java
/**
 * Package: com.wwcs
 * Description: This class contains a method that calculates the power of a string.
 */
package com.wwcs;

import java.util.*;

public class Solution {
    /**
     * Calculate the power of a string.
     * 
     * @param s the input string
     * @return the power of the string
     */
    public int maxPower(String s) {
        int maxPower = 0;
        char currentChar = s.charAt(0);
        int currentCount = 1;

        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == currentChar) {
                currentCount++;
            } else {
                maxPower = Math.max(maxPower, currentCount);
                currentChar = s.charAt(i);
                currentCount = 1;
            }
        }

        return Math.max(maxPower, currentCount);
    }
}

File: updated_code/code/python/1446-consecutive-characters.py
Python
class Solution:
    """
    This class is used to solve the problem of finding the maximum power of a string.
    
    Attributes:
    None
    """

    def max_power(self, s: str) -> int:
        """
        This method calculates the maximum power of a string.
        
        Parameters:
        s (str): The input string.
        
        Returns:
        int: The maximum power of the string.
        """
        max_power = 0
        i = 0
        if not s:  # Handle empty string
            return 0
        while i < len(s):
            count = 1
            while i + 1 < len(s) and s[i] == s[i + 1]:
                i += 1
                count += 1
            max_power = max(max_power, count)
            i += 1
        return max_power